src/Controller/Component/CookieComponent.php

```diff
@@ -246,7 +246,7 @@ protected function _load($key)
         if (isset($this->_loaded[$first])) {
             return;
         }
-        $cookie = $this->request->getCookie($first);
```

Comment:
I don't think removing calls to `$this->request` need to be removed. Requiring access to request/response instance in components is pretty common and those properties should be retained in components, albeit with visibility protected.

Reply:
One issue with components retaining references to the request is that it makes it much easier to make errors with immutability. For example any `withX()` method called on a request held by a component will not update other components or the controller. If we centralize the request on to the controller people will be less likely to make this kind of error.

Reply:
Okay makes sense.